src/DevAdventCalendarCompetition/DevAdventCalendarCompetition.Tests/IntegrationTests/IntegrationTestBase.cs

```diff
@@ -0,0 +1,64 @@
+using System;
+using DevAdventCalendarCompetition.Repository.Context;
+using Microsoft.Data.Sqlite;
+using Microsoft.EntityFrameworkCore;
+
+namespace DevAdventCalendarCompetition.Tests.IntegrationTests
+{
+    public class IntegrationTestBase : IDisposable
+    {
+        private readonly SqliteConnection _connection;
+
+        protected IntegrationTestBase()
+        {
+            var connectionStringBuilder =
+                new SqliteConnectionStringBuilder { DataSource = ":memory:" };
+            var connectionString = connectionStringBuilder.ToString();
+            this._connection = new SqliteConnection(connectionString);
+            this._connection.Open();
+
+            var builder = new DbContextOptionsBuilder<ApplicationDbContext>();
+            builder.UseSqlite(this._connection);
+
+            this.ContextOptions = builder.Options;
+
+            this.Seed();
+        }
+
+        protected DbContextOptions<ApplicationDbContext> ContextOptions { get; }
+
+        public void Dispose()
+        {
+            this.Dispose(true);
+            GC.SuppressFinalize(this);
+        }
+
+        protected virtual void Dispose(bool disposing)
+        {
+            this._connection.Dispose();
+        }
+
+        private void Seed()
+        {
+            using (var context = new ApplicationDbContext(this.ContextOptions))
+            {
+                context.Database.EnsureCreated();
+
+                context.Database.ExecuteSqlRaw(
+                    "INSERT INTO AspNetRoles ([Id],[ConcurrencyStamp],[Name],[NormalizedName]) " +
+                    "VALUES('1','57c244a4-9d0a-4b58-9997-d1c920ca6a21','Admin',null)");
+
+                context.Database.ExecuteSqlRaw(
+                    "INSERT INTO AspNetUsers ([Id],[AccessFailedCount],[ConcurrencyStamp],[Email],[EmailConfirmed],[LockoutEnabled],[LockoutEnd],[NormalizedEmail],[NormalizedUserName]," +
```

Comment:
do we need raw sql here? :(

Reply:
Otherwise, it will be really hard to seed the app user into the database.

Reply:
I meant something like that:
```
var commandText = "INSERT Categories (CategoryName) VALUES (@CategoryName)";
    var name = new SqlParameter("@CategoryName", "Test");
    context.Database.ExecuteSqlCommand(commandText, name);
```

Reply:
Ok, I just didn't want that method to become large by introducing many variables, which won't change often or almost never.

Reply:
We can afford it, it's only one file - it will be more understandable :)